### PR TITLE
INN 2704 Vercel manual resync

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
- Allow Vercel users to manually resync their apps
- Add a new Switch component that uses Radix
- Some design refresh tweaks from John in Alerts and Banners

Screenshot 2024-02-09 at 15 28 33

Next PRs
- Add scrollable content to Modal
- Replace old Switch component in other locations
- Move Alert and Banner components to shared package

## Motivation
- There's a need for manual resync if Vercel users are not using the Inngest integration, or if for some reason the automatic sync fails.

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [X] I've linked any associated issues to this PR.
- [x] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/ResyncModal.tsx [hunk 1]
```diff
@@ -68,45 +69,53 @@ export default function ResyncModal({ isOpen, onClose, url }: Props) {
...
-                title="Override"
-              />
-            </Switch.Group>
-          </div>
+            <SwitchLabel htmlFor="override">Override Input</SwitchLabel>
+          </SwitchWrapper>
+          {isURLOverridden && !failure && (
+            <p className="pl-[50px] pt-1 font-semibold text-yellow-700">
+              Ensure your app ID in the new endpoint is the same, otherwise Inngest will consider it
+              a new app while syncing.
```
Comment (djfarrelly) [inline ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/ResyncModal.tsx:115]:
```suggestion
              Please ensure that your app ID (
              <Link showIcon={true} href="https://www.inngest.com/docs/apps#apps-in-sdk">docs</Link>
              ) is not changed before resyncing. Changing the app ID will result in the creation of a new app in this environment.
```

File: ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/ResyncModal.tsx [hunk 2]
```diff
@@ -68,45 +69,53 @@ export default function ResyncModal({ isOpen, onClose, url }: Props) {
       }
     >
       <div className="border-b border-slate-200 px-6">
+        {platform === 'vercel' && !failure && (
+          <Alert className="my-2" severity="info" showIcon={false}>
```
Comment (djfarrelly) [inline ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/ResyncModal.tsx:73]: suggestion: The top spacing looks a little cramped, should the spacing around this be consistent? Not sure what the exact spacing should be. Please ignore if you already covered this with John!
```suggestion
          <Alert className="my-6" severity="info" showIcon={false}>
```

image

**Review by djfarrelly — CHANGES_REQUESTED** [review]
Updated copy suggestions!

**Comment by goodoldneon** [comment]
LGTM after Dan's suggested changes

**Review by goodoldneon — APPROVED** [review]

**Pull request merged**